pkgs/ffigen/lib/src/header_parser/sub_parsers/api_availability.dart

```diff
@@ -0,0 +1,147 @@
+
+    final alwaysDeprecated = calloc<Int>();
+    final alwaysUnavailable = calloc<Int>();
+    final platforms =
+        calloc<clang_types.CXPlatformAvailability>(platformsLength);
+
+    clang.clang_getCursorPlatformAvailability(cursor, alwaysDeprecated, nullptr,
+        alwaysUnavailable, nullptr, platforms, platformsLength);
+
+    PlatformAvailability? ios;
+    PlatformAvailability? macos;
+
+    for (var i = 0; i < platformsLength; ++i) {
+      final platform = platforms[i];
+      final platformAvailability = PlatformAvailability(
+        introduced: platform.Introduced.triple,
+        deprecated: platform.Deprecated.triple,
+        obsoleted: platform.Obsoleted.triple,
+        unavailable: platform.Unavailable != 0,
+      );
+      switch (platform.Platform.string()) {
+        case 'ios':
+          ios = platformAvailability;
+          break;
+        case 'macos':
+          macos = platformAvailability;
+          break;
+      }
+    }
+
+    final api = ApiAvailability(
+      alwaysDeprecated: alwaysDeprecated.value != 0,
+      alwaysUnavailable: alwaysUnavailable.value != 0,
+      ios: ios,
+      macos: macos,
+    );
+
+    for (var i = 0; i < platformsLength; ++i) {
+      clang.clang_disposeCXPlatformAvailability(platforms + i);
+    }
+    calloc.free(alwaysDeprecated);
+    calloc.free(alwaysUnavailable);
+    calloc.free(platforms);
+
+    return api;
+  }
+
+  bool isAvailable(ObjCTargetVersion minVers) {
+    // If no minimum versions are specified, everything is available.
+    if (minVers.ios == null && minVers.macos == null) {
+      return true;
+    }
+
+    if (alwaysDeprecated || alwaysUnavailable) {
+      return false;
+    }
+
+    for (final (plat, minVer) in [(ios, minVers.ios), (macos, minVers.macos)]) {
+      // If the user hasn't specified a minimum version for this platform, defer
+      // to the other platforms.
+      if (minVer == null) {
+        continue;
+      }
+      // If the API is available on any platform, return that it's available.
+      if (_platformAvailable(plat, minVer)) {
+        return true;
+      }
+    }
+    // The API is not available on any of the platforms the user cares about.
+    return false;
+  }
+
+  bool _platformAvailable(PlatformAvailability? plat, VersionTriple minVer) {
+    if (plat == null) {
+      // Clang's availability info has nothing to say about the given platform,
+      // so assume the API is available.
+      return true;
+    }
+    if (plat.unavailable) {
+      return false;
+    }
+    if (minVer >= plat.deprecated || minVer >= plat.obsoleted) {
+      return false;
+    }
+    return true;
+  }
+
+  @override
+  String toString() => '''Availability {
+  alwaysDeprecated: $alwaysDeprecated
+  alwaysUnavailable: $alwaysUnavailable
+  ios: $ios
+  macos: $macos
+}''';
+}
+
+class PlatformAvailability {
+  VersionTriple? introduced;
+  VersionTriple? deprecated;
+  VersionTriple? obsoleted;
```

Comment:
What is obsoleted? Is it the same as deprecated? E.g. you can still call it?

Reply:
Maybe you can make a suggestion together with @brianquinlan if it makes sense to generated deprecated methods with `@Deprecated` or to completely not generate them based on what you've seen so far in the ObjC/Swift APIs.

Reply:
Ah, you have answered this above. 👍